app/api/route.js

```diff
@@ -0,0 +1,298 @@
+  }
+  type Mutation {
+    addBook(
+      title: String!
+      description: String!
+      author_id: Int!
+      profilePhotoUrl: String
+      publishedAt: String!
+    ): Book
+    addAuthor(
+      name: String!
+      biography: String!
+      profilePhotoUrl: String
+      date_of_birth: String!
+    ): Author
+    updateBook(
+      id: Int!
+      title: String
+      description: String
+      publishedAt: String
+      author_id: Int
+      profilePhotoUrl: String
+    ): Book
+    updateAuthor(
+      id: Int!
+      name: String
+      biography: String
+      date_of_birth: String
+      profilePhotoUrl: String
+    ): Author
+    deleteBook(id: Int!): Int
+    deleteAuthor(id: Int!): Int
+    addReview(bookId: Int!, rating: Int!, comment: String): Book
+  }
+`;
+
+const batchAuthors = new DataLoader(async (author_ids) => {
+  let authorsList = await Author.findAll({
+    where: {
+      id: {
+        [Op.in]: author_ids,
+      },
+    },
+  });
+  authorsList = Promise.all(
+    authorsList.map(async (author) => {
+      return {
+        ...author.toJSON(),
+        profilePhotoUrl: await validateImageUrl(
+          author.profilePhotoUrl,
+          "author"
+        ),
+      };
+    })
+  );
+  const idToAuthor = authorsList.reduce((mapping, author) => {
+    mapping[author.id] = author;
+    return mapping;
+  }, {});
+  return author_ids.map((id) => idToAuthor[id]);
+});
+
+const batchBooks = new DataLoader(async (author_ids) => {
+  let booksList = await Book.findAll({
+    where: {
+      author_id: {
+        [Op.in]: author_ids,
+      },
+    },
+  });
+  booksList = Promise.all(
+    booksList.map(async (book) => {
+      return {
+        ...book.toJSON(),
+
+        profilePhotoUrl: await validateImageUrl(book.profilePhotoUrl, "book"),
+      };
+    })
+  );
+  const authorIdToBook = booksList.reduce((mapping, book) => {
+    if (book.author_id in mapping) {
+      mapping[book.author_id].push(book);
+    } else {
+      mapping[book.author_id] = [book];
+    }
+    return mapping;
+  }, {});
+  return author_ids.map((author_id) =>
+    author_id in authorIdToBook ? authorIdToBook[author_id] : []
+  );
+});
+
+const resolvers = {
+  Query: {
+    books: async (_, { page = 0, size = 5, searchTitle = null }) => {
+      const offset = page * size;
+      const where = searchTitle
+        ? {
+            title: {
+              [Op.iLike]: `%${searchTitle}%`,
+            },
+          }
+        : {};
+
+      const { rows, count } = await Book.findAndCountAll({
+        where,
+        limit: size,
+        offset: offset,
+      });
+
+      const validatedBooks = await Promise.all(
+        rows.map(async (book) => ({
+          ...book.toJSON(),
+          profilePhotoUrl: await validateImageUrl(book.profilePhotoUrl, "book"),
+        }))
+      );
+
+      return {
+        books: validatedBooks,
+        totalCount: count,
+        hasMore: offset + rows.length < count,
+      };
+    },
```

Comment:
As this gets more complex over time, it is good to divide this into two files one which works with graphql and api, other which works with database.